000-vip-init.php

```diff
@@ -265,4 +265,28 @@
 	add_filter( 'wp_sitemaps_enabled', '__return_false' );
 }
 
+// We use HyperDB, so DB_* constants are not automatically defined
+if ( ! defined( 'DB_NAME' ) && ! defined( 'DB_HOST' ) && ! defined( 'DB_PASSWORD' ) && ! defined( 'DB_NAME' ) ) {
+	global $db_servers;
+	if ( ! is_array( $db_servers ) ) {
+		return;
+	}
+
+	$writable_servers = array_filter( $db_servers, function( $db_server ) {
+		return isset( $db_server[5] ) && $db_server[5] > 0;
+	} );
+
+	usort( $writable_servers, function( $a, $b ) {
+		return $a[5] > $b[5];
```

Comment:
```suggestion
		return ( $a[5] ?? 1 ) - ( $b[5] ?? 1 );
```

`??` is because the fifth index, in theory, may not be defined (see above). `-` instead of `>` is because the callback is [expected to return](https://www.php.net/usort) a tri-state integer, not a boolean, value.